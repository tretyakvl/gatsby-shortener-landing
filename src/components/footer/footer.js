import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql, Link } from 'gatsby'

import Logo from '../logo/logo'
import Socials from '../socials/socials'

import style from './footer.module.css'

export const PureFooter = ({ groups, social }) => {
  return (
    <footer className={style.footer}>
      <div className={style.footer__container}>
        <Link to='/'>
          <span className='visually-hidden'>Home</span>
          <Logo type='light' />
        </Link>
        <nav className={style.footer__navigation}>
          {groups.map(({ group }, i) => (
            <section key={i}>
              <h4>{group.title}</h4>
              <ul>
                {group.links.map(([title, to], i) => (
                  <li key={i}>
                    <Link to={to}>{title}</Link>
                  </li>
                ))}
              </ul>
            </section>
          ))}
          <Socials links={social.links} />
        </nav>
      </div>
    </footer>
  )
}

export default props => (
  <StaticQuery
    query={graphql`
      query {
        footerYaml {
          groups {
            group {
              links
              title
            }
          }
          social {
            links
          }
        }
      }
    `}
    render={({ footerYaml }) => <PureFooter {...footerYaml} {...props} />}
  />
)

PureFooter.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.shape({
    group: PropTypes.shape({
      links: PropTypes.arrayOf(PropTypes.array),
      title: PropTypes.string
    })
  })).isRequired,
  social: PropTypes.shape({
    links: PropTypes.arrayOf(PropTypes.array)
  })
}
