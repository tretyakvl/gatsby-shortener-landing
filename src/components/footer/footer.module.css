.footer {
  background-color: var(--gray-700);
}

.footer__container {
  display: flex;
  flex-direction: column;
  align-items: center;

  max-width: var(--max-width);
  margin: var(--margin);
  padding: var(--padding);
  padding-top: 3.2rem;
  padding-bottom: 5rem;

  text-align: center;
}

.footer__container > svg {
  margin-bottom: 3.3rem;
}

.footer__navigation section + section {
  margin-top: 2.6rem;
}

.footer__navigation h4 {
  margin-top: 0;
  margin-bottom: 1.5rem;

  font-size: 1rem;

  color: var(--gray-0);
}

.footer__navigation ul {
  margin: 0;
  padding: 0;

  list-style: none;
}

.footer__navigation a {
  color: var(--gray-400);
  font-size: 0.9rem;
  text-decoration: none;
}

.footer__navigation li + li {
  margin-top: 0.5rem;
}

.footer__navigation > ul {
  margin-top: 2.5rem;
}

@media (min-width: 1500px) {
  .footer__container {
    flex-direction: row;
    align-items: flex-start;

    padding-top: 4.5rem;

    text-align: left;
  }

  .footer__container > svg {
    margin-right: auto;
    margin-bottom: 0;
  }

  .footer__navigation {
    flex: 1 1 0;
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-between;

    max-width: 45rem;
    margin-bottom: 0;
  }

  .footer__navigation section + section {
    margin-top: 0;
  }

  .footer__navigation > ul {
    margin: -0.5rem -0.5rem 0 0;
  }
}
