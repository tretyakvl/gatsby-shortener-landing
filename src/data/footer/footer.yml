logo: ../../images/logo.svg
groups: 
  - group:
      title: 'Features'
      links: 
        - ['Link Shortening', '#']
        - ['Branded Links', '#']
        - ['Analytics', '#']
  - group:
      title: 'Resources'
      links: 
        - ['Blog', '#']
        - ['Developers', '#']
        - ['Support', '#']
  - group:
      title: 'Company'
      links: 
        - ['About', '#']
        - ['Our Team', '#']
        - ['Careers', '#']
        - ['Contact', '#']
social:
    links:
        - ['facebook', '#']
        - ['twitter', '#']
        - ['pinterest', '#']
        - ['instagram', '#']