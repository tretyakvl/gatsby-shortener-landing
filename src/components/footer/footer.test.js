import React from 'react'
import renderer from 'react-test-renderer'
import { PureFooter as Footer } from './footer'

describe('Footer', () => {
  it('renders correctly', () => {
    const props = {
      groups: [
        {
          group: {
            links: [
              ['We', '/we'],
              ['All', '/all'],
              ['Lift', '/lift']
            ],
            title: 'Together'
          }
        },
        {
          group: {
            links: [
              ['check', '/check'],
              ['your', '/your'],
              ['pulse', '/pulse']
            ],
            title: 'It may be gone'
          }
        },
        {
          group: {
            links: [
              ['Call', '/call'],
              ['Your', '/your'],
              ['Grandma', '/grandma']
            ],
            title: 'Regularly'
          }
        }
      ],
      social: {
        links: [
          ['Zuccs', '/purgatory'],
          ['Trees', '/are-bend'],
          ['Cancel', '/everyone']
        ]
      }
    }

    const tree = renderer.create(<Footer {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
