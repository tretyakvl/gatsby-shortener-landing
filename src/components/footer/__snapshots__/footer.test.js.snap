// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__container"
  >
    <a
      href="/"
    >
      <test-file-stub
        className="light"
      />
    </a>
    <nav
      className="footer__navigation"
    >
      <section>
        <h4>
          Together
        </h4>
        <ul>
          <li>
            <a
              href="/we"
            >
              We
            </a>
          </li>
          <li>
            <a
              href="/all"
            >
              All
            </a>
          </li>
          <li>
            <a
              href="/lift"
            >
              Lift
            </a>
          </li>
        </ul>
      </section>
      <section>
        <h4>
          It may be gone
        </h4>
        <ul>
          <li>
            <a
              href="/check"
            >
              check
            </a>
          </li>
          <li>
            <a
              href="/your"
            >
              your
            </a>
          </li>
          <li>
            <a
              href="/pulse"
            >
              pulse
            </a>
          </li>
        </ul>
      </section>
      <section>
        <h4>
          Regularly
        </h4>
        <ul>
          <li>
            <a
              href="/call"
            >
              Call
            </a>
          </li>
          <li>
            <a
              href="/your"
            >
              Your
            </a>
          </li>
          <li>
            <a
              href="/grandma"
            >
              Grandma
            </a>
          </li>
        </ul>
      </section>
      <ul
        className="socials"
      >
        <a
          href="/purgatory"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="visually-hidden"
          >
            Zuccs
          </span>
          <test-file-stub />
        </a>
        <a
          href="/are-bend"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="visually-hidden"
          >
            Trees
          </span>
          <test-file-stub />
        </a>
        <a
          href="/everyone"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="visually-hidden"
          >
            Cancel
          </span>
          <test-file-stub />
        </a>
      </ul>
    </nav>
  </div>
</footer>
`;
